gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -512,7 +518,12 @@ function roleselection.GetSelectableRolesList(maxPlys, rolesAmountList)
 			selectableRoles[subrole] = rolesAmountList[subrole]
 
 			curRoles = curRoles + 1
+
+			break
 		end
+
+		-- If got to this point remove the subrole Table
+		table.remove(availableSubRolesTbl, currentBaserole)
```

Comment:
You don't want to remove the base role from the available sub roles table, because you want the ability to potentially loop through the sub roles of this base role again. This provides the breadth of the randomness

Reply:
Yes I want to remove it, but not in this case, you are right.

I forgot to put a continue statement over it, thanks :)